Without this patch, GCC places all functions declared afterwards in this
file in the .dtors section. This section is not executable, thus
attempting to execute data defined in this section will cause a
segmentation fault. This patch modifies the assembly to ensure that the
declared functions end up in the executable text segment.

See https://gitlab.alpinelinux.org/alpine/aports/-/issues/13012#note_180302

diff -upr pcc-libs-20210420.orig/csu/linux/crtbegin.c pcc-libs-20210420/csu/linux/crtbegin.c
--- pcc-libs-20210420.orig/csu/linux/crtbegin.c	2021-09-16 12:23:33.007528786 +0200
+++ pcc-libs-20210420/csu/linux/crtbegin.c	2021-09-16 12:24:01.290921849 +0200
@@ -56,6 +56,7 @@ asm(	"	.section .ctors,\"aw\",@progbits\
 #else
 	"	.long -1\n"
 #endif
+	"	.section .text\n"
 );
 
 
